src/dnscap.c

```diff
@@ -2957,3 +2962,55 @@ daemonize(void)
 #endif
   logerr("Backgrounded as pid %u", getpid());
 }
+
+static ssize_t
+gzip_cookie_write(void *cookie, const char *buf, size_t size) {
+	return gzwrite((gzFile)cookie, (const void *)buf, (unsigned) size);
```

Comment:
This won't work if there is no zlib.h, maybe do `#ifdef HAVE_LIBZ_H` for all the new code and use the old if it does not exist.